pkg/controller/sparkapplication/controller.go

```diff
@@ -324,13 +332,19 @@ func (c *Controller) getAndUpdateAppState(app *v1beta1.SparkApplication) error {
 			if app.Status.TerminationTime.IsZero() && !currentDriverState.completionTime.IsZero() {
 				app.Status.TerminationTime = currentDriverState.completionTime
 			}
+			// Expose any errors to the users.
+			if currentDriverState.err != nil {
+				app.Status.AppState.ErrorMessage += currentDriverState.err.Error()
+			}
 		}
 		app.Status.AppState.State = newState
 	} else {
 		glog.Warningf("driver not found for SparkApplication: %s/%s", app.Namespace, app.Name)
 		// The application has not terminated and has a recorded driver Pod, but no driver Pod was found for it.
 		// This is likely because the driver Pod was deleted. In this case, set the application state to FailingState.
-		if app.Status.TerminationTime.IsZero() && app.Status.DriverInfo.PodName != "" {
+		// Also move to FailingState if we never recorded the driver for a submitted application and it has been a while => 300 seconds
+		if app.Status.TerminationTime.IsZero() && (app.Status.DriverInfo.PodName != "" ||
+			(!app.Status.LastSubmissionAttemptTime.IsZero() && metav1.Now().After(app.Status.LastSubmissionAttemptTime.Add(time.Duration(300)*time.Second)))) {
```

Comment:
in the case on a non job scenario - we should probably halt immediately as this should be a sync creation of a pod